addons/mrp/models/stock_move.py

```diff
@@ -83,6 +83,26 @@ def _get_aggregated_properties(self, move_line=False, move=False):
         aggregated_properties['line_key'] += f'_{bom.id if bom else ""}'
         return aggregated_properties
 
+    def _compute_product_packaging_qty(self):
+        # we catch kit lines where the packaging is still the one from the final product (it has
+        # not been change to a packaging of a component)
+        kit_lines = self.filtered(lambda move_line: move_line.move_id.bom_line_id.bom_id.type == 'phantom' and
+        move_line.product_id != move_line.move_id.product_packaging_id.product_id)
+        for move_line in kit_lines:
+            move = move_line.move_id
+            bom_line = move.bom_line_id
+            kit_bom = bom_line.bom_id
+
+            # Convert the move line quantity to the product's move uom
+            qty_move_uom = move_line.product_uom_id._compute_quantity(move_line.quantity, move_line.move_id.product_uom)
+            # Convert the product's move uom to the bom line's uom
+            qty_bom_uom = move.product_uom._compute_quantity(qty_move_uom, bom_line.product_uom_id)
+            # calculate the bom's kit qty in kit product uom qty
+            bom_qty_product_uom = kit_bom.product_uom_id._compute_quantity(kit_bom.product_qty, kit_bom.product_tmpl_id.uom_id)
+            # calculate the quantity needed of packging
+            move_line.product_packaging_qty = (qty_bom_uom / (bom_line.product_qty / bom_qty_product_uom)) / move_line.move_id.product_packaging_id.qty
```

Comment:
```suggestion
            qty_of_kits = qty_bom_uom / bom_line.product_qty
            qty_of_kits_in_one_packaging = bom_qty_product_uom / move_line.move_id.product_packaging_id.qty
            move_line.product_packaging_qty = qty_of_kits / qty_of_kits_in_one_packaging
```
I wonder if it shouldn't be broken down into more lines to make this understandable :confused: 
Because I find it *very* confusing, especially with parenthesis wrapping around terms that have different meanings.